docker-app/qfieldcloud/core/tests/test_jobs.py

```diff
@@ -0,0 +1,203 @@
+
+        # Create a project
+        self.p1 = Project.objects.create(name="p1", is_public=False, owner=self.u1)
+
+    def upload_file(
+        self,
+        project: Project,
+        local_filename: str,
+        remote_filename: str,
+    ) -> Response:
+        """Upload a file to QFieldCloud using API.
+
+        Args:
+            project (Project): project that should contain the file.
+            local_filename (str): name of the local file to upload, should be in `testdata` folder.
+            remote_filename (str): name of the uploaded file.
+
+        Returns:
+            Response: response to the POST HTTP request.
+        """
+        file_path = testdata_path(local_filename)
+        response = self.client.post(
+            f"/api/v1/files/{project.id}/{remote_filename}/",
+            {
+                "file": open(file_path, "rb"),
+            },
+            format="multipart",
+        )
+        return response
+
+    def test_bad_layer_handler_values_for_process_projectfile_job(self):
+        # Test that BadLayerHandler is parsing data properly during process projectfile jobs
+        self.client.credentials(HTTP_AUTHORIZATION="Token " + self.t1.key)
+
+        # Push the QGIS project and datasource GPKG files
+        response = self.upload_file(self.p1, "bumblebees.gpkg", "bumblebees.gpkg")
+
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+        response = self.upload_file(
+            self.p1, "simple_bumblebees_wrong_localized.qgs", "project.qgs"
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+        wait_for_project_ok_status(self.p1)
+        processprojectfile_job = Job.objects.filter(
+            project=self.p1, type=Job.Type.PROCESS_PROJECTFILE
+        ).latest("updated_at")
+
+        self.assertEqual(processprojectfile_job.status, Job.Status.FINISHED)
+        self.assertIsNotNone(processprojectfile_job.feedback)
+
+        self.p1.refresh_from_db()
+
+        self.assertIsNotNone(self.p1.project_details)
+
+        # extract layer data from job
+        processfile_layers = processprojectfile_job.get_feedback_step_data(
+            "project_details"
+        )["returns"]["project_details"]["layers_by_id"]
+
+        # "valid" localized layer -> QFC considers it as invalid
+        valid_localized_layer = processfile_layers["valid_localized_point_layer_id"]
+
+        self.assertFalse(valid_localized_layer["is_valid"])
+        self.assertTrue(valid_localized_layer["is_localized"])
+
+        # invalid localized layer
+        invalid_localized_layer = processfile_layers[
+            "invalid_localized_polygon_layer_id"
+        ]
+
+        self.assertFalse(invalid_localized_layer["is_valid"])
+        self.assertTrue(invalid_localized_layer["is_localized"])
+        self.assertEquals(
+            invalid_localized_layer["error_code"], "localized_dataprovider"
+        )
```

Comment:
If you define:
```
    def assertLayerData(self, layer_data: dict, is_valid: bool, is_localized: bool, error_code: str):
        self.assertEquals(layer_data["is_valid"], is_valid)
        self.assertEquals(layer_data["is_localized"], is_localized)
        self.assertEquals(layer_data["error_code"], error_code)
```
You can simplify the code and improve readability to something like:

```suggestion
        self.assertLayerData(
	        processfile_layers["invalid_localized_polygon_layer_id"],
	        False,
	        True,
	        "localized_dataprovider",
        )
```